src/status_im/contexts/wallet/add_account/create_account/key_pair_name/view.cljs

```diff
@@ -0,0 +1,94 @@
+(ns status-im.contexts.wallet.add-account.create-account.key-pair-name.view
+  (:require
+    [clojure.string :as string]
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [status-im.common.floating-button-page.view :as floating-button-page]
+    [status-im.common.not-implemented :as not-implemented]
+    [status-im.common.validation.general :as validators]
+    [status-im.contexts.wallet.add-account.create-account.key-pair-name.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def ^:private key-pair-name-max-length 15)
+(def ^:private key-pair-name-min-length 5)
+
+(def error-messages
+  {:too-long     (i18n/label :t/key-name-error-length)
+   :too-short    (i18n/label :t/key-name-error-too-short {:count key-pair-name-min-length})
+   :emoji        (i18n/label :t/key-name-error-emoji)
+   :special-char (i18n/label :t/key-name-error-special-char)})
+
+(defn navigate-back
+  []
+  (rf/dispatch [:navigate-back]))
+
+(defn view
+  []
+  (let [{:keys [workflow]}                (rf/sub [:get-screen-params])
+        customization-color               (rf/sub [:profile/customization-color])
+        [key-pair-name set-key-pair-name] (rn/use-state "")
+        [error set-error]                 (rn/use-state nil)
+        on-change-text                    (rn/use-callback
+                                           (fn [value]
+                                             (set-key-pair-name value)
+                                             (cond
+                                               (> (count value) key-pair-name-max-length)
+                                               (set-error :too-long)
```

Comment:
Hum, I think we can separate side-effectful code from pure code.

My suggestion here is to define a pure function `validate-key-pair-name` as a separate var that returns the type of error or nil. Then, your `on-change-text` will be just two nice lines:

```clojure
(rn/use-callback
 (fn [value]
   (set-key-pair-name value)
   (set-error (validate-key-pair-name value))))
```

After this new function is defined, I would unit test it in a file named `view_test.cljs`.

Another interesting option is to completely extract the validation part from the `view` namespace and move them to a `validation.cljs` file inside `key_pair_name/`. That way the design sort of forces developers to keep pure code more separated from side-effects and also creates a stronger incentive to unit test the behavior because in view namespaces there's a strong tendency to not unit test anything there.